src/plugins/intel_gpu/src/graph/impls/ocl/rope.cpp

```diff
@@ -44,15 +44,16 @@ struct rope_impl : typed_primitive_impl_ocl<rope> {
         params.slice_start = primitive->config.slice_start;
         params.slice_stop = primitive->config.slice_stop;
 
-        params.axis = primitive->config.is_qwen || primitive->config.is_chatglm ? 2 : 3;
-        params.num_of_inputs = primitive->config.is_chatglm || primitive->config.is_interleaved ? 2 : 3;
+        params.axis = primitive->config.is_qwen  || primitive->config.is_chatglm || primitive->config.support_2d_rope ? 2 : 3;
```

Comment:
Do we need this check? 
Currently support_2d_rope is only implemented for is_chatglm. 
Which means that if 2d_rope == true than is_chatglm == true. 

Also maybe somewhre we need to add an assertion check that if support_2d_rope then is_chatglm is true.

Reply:
Yes, you're right. it's redundant code, I missed to remove them after refactoring. I deleted it. thanks.

Reply:
>> Also maybe somewhre we need to add an assertion check that if support_2d_rope then is_chatglm is true.

Please apply this comment too

Reply:
applied